packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx

```diff
@@ -110,7 +109,16 @@ interface PresentationInstanceFilterDialogContentProps extends Omit<Presentation
 
 function PresentationInstanceFilterDialogContent(props: PresentationInstanceFilterDialogContentProps) {
   const { onApply, initialFilter, descriptor, imodel, ruleGroupDepthLimit, filterResultsCountRenderer, onClose, descriptorInputKeys } = props;
-  const [initialPropertyFilter] = useState(() => (initialFilter ? convertPresentationFilterToPropertyFilter(descriptor, initialFilter.filter) : undefined));
+  const [initialPropertyFilter] = useState(() => {
+    if (!initialFilter) {
+      return undefined;
+    }
+    try {
+      return PresentationInstanceFilter.toComponentsPropertyFilter(descriptor, initialFilter.filter);
+    } catch {}
```

Comment:
Why hide the errors? Maybe the component should capture the error state and render `<NonIdealState />` in such cases?

Reply:
Added ErrorBoundary around filter dialog content